### PR TITLE
[fix](jdbc catalog) fix bugs of jdbc catalog and table valued function

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx
This pr fix some bugs:
1. `desc function tvf(..)` throws NullPointException.
2. `Show Tables` in multi catalog did not catch some exception.
3. jdbc catalog did not handle `decimal unsigned` mysql data type.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.88 seconds
 load time: 635 seconds
 storage size: 17123274847 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221220152152_clickbench_pr_65793.html

File: fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java [hunk 1]
```diff
@@ -338,6 +341,14 @@ public Type mysqlTypeToDoris(JdbcFieldSchema fieldSchema) {
                     return Type.BIGINT;
                 case "BIGINT":
                     return ScalarType.createStringType();
+                case "DECIMAL":
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java:344]: Please add regression test for this case

Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java:345]: done

**Review by morningman — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by zhangguoqiang666 — COMMENTED** [review]
GOOD

**Review by zhangguoqiang666 — COMMENTED** [review]
P0 external hahah P1  for test

**Pull request merged**